chutney/engine/src/main/java/com/chutneytesting/engine/domain/execution/strategies/RetryWithTimeOutStrategy.java

```diff
@@ -106,6 +113,19 @@ public Status execute(ScenarioExecution scenarioExecution,
         return st;
     }
 
+    private Object evaluateExpression(String expression, StepDataEvaluator evaluator, ScenarioContext scenarioContext) {
```

Comment:
you dont need this method which is redondant with StepDataEvaluator. Just call evaluator.evaluate(expression, scenarioContext)

Reply:
Be careful here, the context to use to evaluate the expression should be the merge context from `scenarioContext` and `localContext` i think, as this strategy could be used into a for strategy loop.

Reply:
Maybe it could be cool that this strategy put a `retryCount` variable into the context that could be used.